api/repositories/buildpack_repository.go

```diff
@@ -56,11 +57,22 @@ func (r *BuildpackRepository) ListBuildpacks(ctx context.Context, authInfo autho
 	)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			return nil, fmt.Errorf("no BuilderInfo %q resource found in %q namespace", r.builderName, r.rootNamespace)
+			return nil, apierrors.NewResourceNotFoundError(fmt.Errorf("BuilderInfo %q not found in namespace %q", r.builderName, r.rootNamespace))
 		}
+
 		return nil, apierrors.FromK8sError(err, BuildpackResourceType)
 	}
 
+	if !meta.IsStatusConditionTrue(builderInfo.Status.Conditions, StatusConditionReady) {
+		conditionNotReadyMessage := "Resource not reconciled"
+		readyCondition := meta.FindStatusCondition(builderInfo.Status.Conditions, StatusConditionReady)
+		if readyCondition != nil {
+			conditionNotReadyMessage = readyCondition.Message
```

Comment:
We should probably check if the condition's message is empty before overwriting the default message

Reply:
Fixed.